src/Libraries/RevitNodes/Elements/TextNote.cs

```diff
@@ -0,0 +1,280 @@
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        /// <summary>
+        /// Set Text
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="type"></param>
+        /// <param name="location"></param>
+        /// <param name="alignment"></param>
+        /// <param name="angle"></param>
+        private void InternalSetType(string value, RVT.TextNoteType type, XYZ location, HorizontalTextAlignment alignment, double angle)
+        {
+            int bold = type.LookupParameter("Bold").AsInteger();
+            int italic = type.LookupParameter("Italic").AsInteger();
+            string font = type.LookupParameter("Text Font").AsString();
+            double size = type.LookupParameter("Text Size").AsDouble();
+            bool isBold = (bold == 1) ? true : false;
+            bool isItalic = (italic == 1) ? true : false;
+
+            InternalSetTextSettings(value, isBold, isItalic, size, font, location.ToPoint(true), alignment, angle);
+        }
+
+        /// <summary>
+        /// Init element from existing
+        /// </summary>
+        /// <param name="element"></param>
+        private void InitElement(Autodesk.Revit.DB.TextNote element)
+        {
+            LocationPoint location = element.Location as LocationPoint;
+            InternalSetType(element.Text, element.TextNoteType, location.Point, element.HorizontalAlignment, location.Rotation.ToDegrees());
+            InternalSetElement(element);
+        }
+
+        /// <summary>
+        /// Init element by location
+        /// </summary>
+        /// <param name="view"></param>
+        /// <param name="origin"></param>
+        /// <param name="alignment"></param>
+        /// <param name="text"></param>
+        /// <param name="keepRotatedTextreadable"></param>
+        /// <param name="rotation"></param>
+        /// <param name="typeId"></param>
+        private void Init(RVT.View view, RVT.XYZ origin, RVT.HorizontalTextAlignment alignment, string text, bool keepRotatedTextreadable, double rotation, ElementId typeId)
+        {
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+            var element = ElementBinder.GetElementFromTrace<RVT.TextNote>(document);
+
+            RVT.TextNoteOptions options = new TextNoteOptions()
+            {
+                HorizontalAlignment = alignment,
+                KeepRotatedTextReadable = keepRotatedTextreadable,
+                Rotation = rotation.ToRadians(),
+                TypeId = typeId
+            };
+
+            if (element == null)
+            {
+                element = RVT.TextNote.Create(document, view.Id, origin, text, options);
+            }
+            else
+            {
+                element.HorizontalAlignment = alignment;
```

Comment:
again I think checking these properties and setting them only if they have changed is good practice.